utils/bot/bot.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package bot
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+type Bot interface {
```

Comment:
The Bot interface has the basic things a bot can do, which is mainly Ensure its existence, and have a userID. It does not much by itself, but I consider it a basic building block.

In MSCalendar, for example, you build up your bot that implements:
- Bot
- Logger
- Poster
- Admin
- FlowController
- Welcomer (an internal interface from MSCalendar)

If we were not to have the Bot interface, we would have to deal directly with the API to Ensure the bot, store the ID...
Putting this functionality in other components (like poster or logger) would make them less generic (right now they do not need to depend on a bot) and cause repetition (I would have to ensure the logger and the poster separately).

Reply:
Understood! Will different name make more sense? Maybe `Ensurer`, I'm 0/5

Reply:
In terms of functionality, it makes sense to call it Ensurer, but in terms of concept, I still prefer bot, since it is "what it is", instead of "what it does". Nevertheless, I am also 0/5 on this.